src/imageoverlay.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) QuantStack
+// Distributed under the terms of the Modified BSD License.
+import { DOMWidgetModel, ISerializers } from '@jupyter-widgets/base';
+import { BaseOverlayModel, BaseOverlayView } from './baseoverlay';
+
+import 'ol/ol.css';
+import { MODULE_NAME, MODULE_VERSION } from './version';
+import '../css/widget.css';
+
+export class ImageOverlayModel extends BaseOverlayModel {
+  defaults() {
+    return {
+      ...super.defaults(),
+      _model_name: ImageOverlayModel.model_name,
+      _model_module: ImageOverlayModel.model_module,
+      _model_module_version: ImageOverlayModel.model_module_version,
+      _view_name: ImageOverlayModel.view_name,
+      _view_module: ImageOverlayModel.view_module,
+      _view_module_version: ImageOverlayModel.view_module_version,
+      overlay_type: 'image',
+      image_url: '',
+    };
+  }
+
+  static serializers: ISerializers = {
+    ...DOMWidgetModel.serializers,
+    // Ajoutez ici tous les sérialiseurs supplémentaires
+  };
+
+  static model_name = 'ImageOverlayModel';
+  static model_module = MODULE_NAME;
+  static model_module_version = MODULE_VERSION;
+  static view_name = 'ImageOverlayView';
+  static view_module = MODULE_NAME;
+  static view_module_version = MODULE_VERSION;
+}
+
+export class ImageOverlayView extends BaseOverlayView {
+  render() {
+    this.element = document.createElement('img');
+    super.render();
```

Comment:
That works, but the rule should be to always call `super` first.

I am actually surprised TypeScript is not complaining here.

The problem is that doing so would break your widget because `this.element` would not be defined when creating the overlay.

One approach could be to define an `abstract createElement` method in `BaseOverlayView`, and all subclasses would need to provide the implementation., then you can call `createElement` in your `BaseOverlayView`'s `createOverlay` method